lib/runner.js

```diff
@@ -574,13 +574,28 @@ internals.protect = function (item, state) {
         /* $lab:coverage:off$ */
         const promiseRejectionHandler = function (err) {
 
+            if (flags.onUnhandledRejection) {
+                flags.onUnhandledRejection(err);
+                return;
+            }
+
             failedWithUnhandledRejection = true;
             finish(err, 'unhandledRejection');
         };
         process.on('unhandledRejection', promiseRejectionHandler);
 
         const processUncaughtExceptionHandler = function (err) {
 
+            if (flags.onUncaughtException) {
+                try {
+                    flags.onUncaughtException(err);
+                    return;
+                }
+                catch (errInErrorhandling) {
+                    err = errInErrorhandling;
```

Comment:
I figure if the user has `onUncaughtException` declared, then they know what they expect as an `err`, so it's best to fail the test with whatever happens inside of it, rather than with the original error.

Reply:
This sounds like a good plan to me